xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiURLFactory.java

```diff
@@ -169,6 +185,8 @@ URL createAttachmentRevisionURL(String filename, String spaces, String name, Str
      */
     String getURL(URL url, XWikiContext context);
 
+    String getURL(URL url, XWikiContext context, Map<String, String> queryParameters);
```

Comment:
The point of XWikiServletURLFactory#getURL is not to generate an URL but to move from absolute to relative URL (if applicable). It's really not its job to add a query string the way it's currently defined.